### PR TITLE
Profile uniform naming

### PR DESCRIPTION
**Pull request by JasonTuTW**
- "competitionCategory" is changed to "group"
- "organization" is changed to "institution", and a table records the institution id and its name
- change the format of the returned data
- endpoints for institution info

### CODE REVIEW

File: backend/internal/response/reponse.go
Comment (lolainta) [inline backend/internal/response/reponse.go:1]: typo on filename? 
response/reponse

File: backend/api/user.go [hunk 1]
```diff
@@ -15,13 +15,13 @@ import (
 // @Tags         user
 // @Accept       json
 // @Produce      json
-// @Param   	 username 	  formData string true "user's name"
-// @Param   	 password 	  formData string true "password"
-// @Param   	 email 	  	  formData string true "email"
-// @Param   	 overview 	  formData string false "overview"
-// @Param   	 organization formData string false "organization"
+// @Param   	 username 	  	formData string true "user's name"
```
Comment (lolainta) [inline backend/api/user.go]: Use only tabs or only space for alignments

File: backend/api/user.go [hunk 2]
```diff
@@ -53,7 +53,17 @@ func Register(c *gin.Context) {
 		return
 	}
 	user.Overview = c.PostForm("overview")
-	user.Organization = c.PostForm("organization")
+	insIDStr := c.PostForm("institutionID")
+	if insIDStr == "" {
+		user.InstitutionID = 1
```
Comment (lolainta) [inline backend/api/user.go]: For this default value, can you add it to docs?

File: backend/api/user.go [hunk 3]
```diff
@@ -15,13 +15,13 @@ import (
...
-// @Param   	 username 	  formData string true "user's name"
-// @Param   	 password 	  formData string true "password"
-// @Param   	 email 	  	  formData string true "email"
-// @Param   	 overview 	  formData string false "overview"
-// @Param   	 organization formData string false "organization"
+// @Param   	 username 	  	formData string true "user's name"
+// @Param   	 password 	  	formData string true "password"
+// @Param   	 email 	  	  	formData string true "email"
+// @Param   	 overview 	  	formData string false "overview"
+// @Param   	 institutionID 	formData string false "institution ID"
```
Comment (lolainta) [inline backend/api/user.go]: I think it is possible for informing the default value.
https://github.com/swaggo/swag#attribute

File: backend/api/user.go (hunk 1 above)
Comment (lolainta) [reply backend/api/user.go]: Remember to check it for all files under api folder

**Review by lolainta — APPROVED** [review]

File: backend/api/user.go (hunk 2 above)
Comment (JasonTuTW) [reply backend/api/user.go]: For now, I choose to add a institution named "no institution", so I will delete the default value, and the field "institutionID" is required.

File: backend/api/user.go (hunk 3 above)
Comment (JasonTuTW) [reply backend/api/user.go]: As mentioned above.

**Pull request merged**